terraform/environments/data-platform/iam.tf

```diff
@@ -357,29 +541,24 @@ data "aws_iam_policy_document" "iam_policy_document_for_reload_data_product_lamb
 }
 
 data "aws_iam_policy_document" "iam_policy_document_for_resync_unprocessed_files_lambda" {
+  source_policy_documents = [data.aws_iam_policy_document.log_to_bucket.json, data.aws_iam_policy_document.read_metadata.json]
+
   statement {
-    sid       = "ListBucket"
-    effect    = "Allow"
-    actions   = ["s3:ListBucket"]
-    resources = [module.s3-bucket.bucket.arn, "${module.s3-bucket.bucket.arn}/*"]
+    sid     = "ListBucket"
+    effect  = "Allow"
+    actions = ["s3:ListBucket"]
+    resources = [
+      module.data_s3_bucket.bucket.arn,
+      "${module.data_s3_bucket.bucket.arn}/raw/*",
+    "${module.data_s3_bucket.bucket.arn}/curated/*"]
```

Comment:
This is a ListBucket operation, so only functions on the bucket object, not these key paths